src/debug/debugInfoProvider.ts

```diff
@@ -0,0 +1,27 @@
+/*-----------------------------------------------------------------------------------------------
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the EPL v2.0 License. See LICENSE file in the project root for license information.
+ *-----------------------------------------------------------------------------------------------*/
+
+import { DebugInfo } from './debugInfo';
+import { Protocol, RSPClient } from 'rsp-client';
+
+export class DebugInfoProvider {
+    public static async retrieve(server: Protocol.ServerHandle, client: RSPClient): Promise<DebugInfo> {
+        const launchCommand: Protocol.CommandLineDetails = await client.getOutgoingHandler().getLaunchCommand({
+            mode: 'debug',
+            params: {
+                id: server.id,
+                serverType: server.type.id,
+                attributes: undefined
+            }
+        });
+
+        return this.create(launchCommand);
```

Comment:
maybe we should check launchCommand before to call the create method. Not sure if getLaunchCommand can return undefined or null